unityfiles/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlaySoundEventArgs : EventArgs {
+}
+
+public class AudioManager : Singleton<AudioManager> {
+    
+    public Sound[] sounds;
+    
+    private void Awake() {
+        SubscribeToEvents();
+        
+        DontDestroyOnLoad(gameObject);
+        
+        foreach (Sound s in sounds) {
+            s.audioSource = gameObject.AddComponent<AudioSource>();
+            s.audioSource.clip = s.AudioClip;
+            
+            // So that we dont have to manually write the name of the songs
+            s.Name = s.audioSource.clip.name;
+
+            s.audioSource.volume = s.Volume;
+            s.audioSource.pitch = s.Pitch;
+            s.audioSource.loop = s.Loop;
+            s.audioSource.mute = s.Mute;
+        }
+    }
+
+    private void SubscribeToEvents() {
+        GameManager.OnMainMenuMusic += CallBackMainMenuMusicEvent;
+        GameManager.OnLevel1Music += CallBackLevel1MusicEvent;
+        Debug.Log("Handlers added");
+    }
+
+    private void UnsubscribeFromEvents() {
+        GameManager.OnMainMenuMusic -= CallBackMainMenuMusicEvent;
+        GameManager.OnLevel1Music -= CallBackLevel1MusicEvent;
+    }
+    
+    private void CallBackMainMenuMusicEvent(object o, EventArgs args) {
+        PlaySound("Main menu music");
+    }
+
+    private void CallBackLevel1MusicEvent(object o, EventArgs args) {
+       PlaySound("Level 1 music");
+       Debug.Log("Play the sound");
+    }
```

Comment:
Tror det vil v;re enklere og bruke enumeratorer eller string consts for og indexe sangene. Du kan foreksempel lage en enumerator som heter songs og putte et felt som heter song index i song da f\r man en dropdown meny hvor man kan velge enumerator til sagene.